### PR TITLE
[TASK] Move "scheduler"-classes into "Classes/Task"

### PR DESCRIPTION
**Pull request by tomasnorre**
Resolved issue #121

### CODE REVIEW

File: Classes/Tasks/CrawlMultiProcessTask.php [hunk 1]
```diff
@@ -24,32 +25,34 @@
  *  This copyright notice MUST APPEAR in all copies of the script!
  ***************************************************************/
 
+use TYPO3\CMS\Scheduler\Task\AbstractTask;
+
 /**
- * Class tx_crawler_scheduler_crawlMultiProcess
- *
- * @package AOE\Crawler\Task
+ * Class CrawlMultiProcessTask
```
Comment (srotsch) [inline Classes/Tasks/CrawlMultiProcessTask.php]: Pls. add empty line below

File: Classes/Tasks/CrawlMultiProcessTaskAdditionalFieldProvider.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ *  This copyright notice MUST APPEAR in all copies of the script!
+ ***************************************************************/
+
+use TYPO3\CMS\Core\Messaging\FlashMessage;
+use TYPO3\CMS\Scheduler\AdditionalFieldProviderInterface;
+use TYPO3\CMS\Scheduler\Controller\SchedulerModuleController;
+use TYPO3\CMS\Scheduler\Task\AbstractTask;
+
+/**
+ * Class CrawlMultiProcessTaskAdditionalFieldProvider
```
Comment (srotsch) [inline Classes/Tasks/CrawlMultiProcessTaskAdditionalFieldProvider.php]: Pls add empty line

File: Classes/Tasks/CrawlerQueueTask.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+ *  GNU General Public License for more details.
+ *
+ *  This copyright notice MUST APPEAR in all copies of the script!
+ ***************************************************************/
+
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Scheduler\Task\AbstractTask;
+
+/**
+ * Class CrawlerQueueTask
```
Comment (srotsch) [inline Classes/Tasks/CrawlerQueueTask.php]: Pls add empty line

File: Classes/Tasks/CrawlerQueueTask.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+     * @return bool
+     */
+    public function execute()
+    {
+        $this->setCliArguments();
+
+        /* @var $crawlerObject \tx_crawler_lib */
+        $crawlerObject = GeneralUtility::makeInstance('tx_crawler_lib');
+        $crawlerObject->CLI_main_im();
+        return TRUE;
```
Comment (srotsch) [inline Classes/Tasks/CrawlerQueueTask.php]: lowercase true

File: Classes/Tasks/CrawlerQueueTaskAdditionalFieldProvider.php [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+ ***************************************************************/
+
+use TYPO3\CMS\Backend\Utility\BackendUtility;
+use TYPO3\CMS\Core\Messaging\FlashMessage;
+use TYPO3\CMS\Scheduler\AdditionalFieldProviderInterface;
+use TYPO3\CMS\Scheduler\Controller\SchedulerModuleController;
+use TYPO3\CMS\Scheduler\Task\AbstractTask;
+
+/**
+ * Class CrawlerQueueTaskAdditionalFieldProvider
```
Comment (srotsch) [inline Classes/Tasks/CrawlerQueueTaskAdditionalFieldProvider.php]: Pls add empty line

File: Classes/Tasks/CrawlerQueueTaskAdditionalFieldProvider.php [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+            'label' => 'LLL:EXT:crawler/locallang_db.xml:crawler_im.conf'
+        );
+
+        return $additionalFields;
+    }
+
+    /**
+     * Mark current value as selected by returning the "selected" attribute
+     *
+     * @access protected
```
Comment (srotsch) [inline Classes/Tasks/CrawlerQueueTaskAdditionalFieldProvider.php]: I'd say these access comments can safely be removed

File: Classes/Tasks/CrawlerTask.php [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  This copyright notice MUST APPEAR in all copies of the script!
+ ***************************************************************/
+
+use TYPO3\CMS\Scheduler\Task\AbstractTask;
+
+/**
+ * Class CrawlerTask
```
Comment (srotsch) [inline Classes/Tasks/CrawlerTask.php]: Empty line pls

You might want to adjust your PHPStorm defaults :-P

File: Classes/Tasks/CrawlerTaskAdditionalFieldProvider.php [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+ *  This copyright notice MUST APPEAR in all copies of the script!
+ ***************************************************************/
+
+use TYPO3\CMS\Core\Messaging\FlashMessage;
+use TYPO3\CMS\Scheduler\AdditionalFieldProviderInterface;
+use TYPO3\CMS\Scheduler\Controller\SchedulerModuleController;
+use TYPO3\CMS\Scheduler\Task\AbstractTask;
+
+/**
+ * Class CrawlerTasksAdditionalFieldProvider
```
Comment (srotsch) [inline Classes/Tasks/CrawlerTaskAdditionalFieldProvider.php]: Guess what :-)

File: Classes/Tasks/CrawlerTaskAdditionalFieldProvider.php [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+ * @package AOE\Crawler\Tasks
+ */
+class CrawlerTasksAdditionalFieldProvider implements AdditionalFieldProviderInterface
```
Comment (srotsch) [inline Classes/Tasks/CrawlerTaskAdditionalFieldProvider.php]: CrawlerTaskAdditionalFieldProvider (singular, file name is correct)

File: Classes/Tasks/FlushQueueTask.php [hunk 10]
```diff
@@ -24,28 +25,31 @@
...
  ***************************************************************/
 
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Scheduler\Task\AbstractTask;
+
 /**
- * Class tx_crawler_scheduler_flush
- *
- * @package AOE\Crawler\Task
+ * Class FlushQueueTask
```
Comment (srotsch) [inline Classes/Tasks/FlushQueueTask.php]: :-D

File: Classes/Tasks/FlushQueueTaskAdditionalFieldProvider.php [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+ *  This copyright notice MUST APPEAR in all copies of the script!
+ ***************************************************************/
+
+use TYPO3\CMS\Scheduler\AdditionalFieldProviderInterface;
+use TYPO3\CMS\Scheduler\Controller\SchedulerModuleController;
+use TYPO3\CMS\Scheduler\Task\AbstractTask;
+
+/**
+ * Class FlushQueueTaskAdditionalFieldProvider
+ * @package AOE\Crawler\Tasks
```
Comment (srotsch) [inline Classes/Tasks/FlushQueueTaskAdditionalFieldProvider.php]: :-D

File: ext_localconf.php [hunk 12]
```diff
@@ -43,30 +43,30 @@
 	)
 );
 
-$GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['scheduler']['tasks']['tx_crawler_scheduler_im'] = array(
+$GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['scheduler']['tasks']['AOE\Crawler\Tasks\CrawlerQueueTask'] = array(
```
Comment (srotsch) [inline ext_localconf.php]: Not 100% sure, but I'd say all backslashes should be escaped to \\

Comment (tomasnorre) [inline ext_localconf.php]: You are right check here: https://github.com/AOEpeople/aoe_ipauth/blob/master/ext_localconf.php#L22 

But don't think is a breaking :) At least not as long as there are no conflicts in the namespace :)

**Comment by tomasnorre** [comment]
Code is updated please review again @srotsch :) 

PS: I have updated my PHPStorm class comment setup ;)

**Pull request merged**